server/app/services/geo/esri/EsriClient.java

```diff
@@ -179,6 +177,35 @@ public CompletionStage<AddressSuggestionGroup> getAddressSuggestions(Address add
             });
   }
 
+  /**
+   * Get the state abbreviation from the attributes json node. on Public ESRI this is under the
+   * RegionAbbr field, but some custom implementations may have it under the Region field.
+   */
+  private String getStateAbbreviationFromAttributes(JsonNode attributes, Address address) {
+    BiFunction<JsonNode, String, String> getValue =
+        (jsonNode, keyName) ->
+            (jsonNode.get(keyName) == null || jsonNode.get(keyName).isNull())
```

Comment:
**question:** Why do we need both of these null checks? Should it be isEmpty like above?

Reply:
One checks if the object is null, the other checks for a null node type. I'm going to write some more tests, I think using isEmpty is actually a bug.

Reply:
@dkatzz, Ok fixed a whole different bug than I planned. Turns out `isEmpty` on scalars always returns true. It's really looking at children in arrays/objects. The `isNull` method is the correct thing to use. I've pull things apart a little more to fix it and switched to a more targeted series of tests to check the mapping.

Reply:
Ah makes sense - thanks for the update! The way you pulled things apart LGTM - just some minor comments.